tiledb/sm/query/reader.cc

```diff
@@ -1536,84 +1536,110 @@ Status Reader::compute_result_coords(
-  // (e.g. unsplittable), it will be corrected in a retry.
-  uint64_t sub_partitioner_memory_budget =
-      constants::sub_partitioner_memory_budget;
-  uint64_t sub_partitioner_memory_budget_var =
-      constants::sub_partitioner_memory_budget;
-
-  // Create a sub-partitioner to partition the current subarray in
-  // `read_state_.partitioner_`. This allows us to compute the range
-  // result coords and subarray coords on a smaller set of elements.
-  // The motiviation for this is primarily to avoid sorting a large
-  // number of elements within a `parallel_sort` because it has a
-  // time complexity of O(N*log(N)).
-  SubarrayPartitioner* const partitioner = &read_state_.partitioner_;
-  SubarrayPartitioner sub_partitioner(
-      partitioner->current(),
-      sub_partitioner_memory_budget,
-      sub_partitioner_memory_budget);
-
-  // Set the individual attribute budgets in the sub-partitioner
-  // to the same values as in the parent partitioner.
-  for (const auto& kv : *partitioner->get_result_budgets()) {
-    const std::string& attr_name = kv.first;
-    const SubarrayPartitioner::ResultBudget& result_budget = kv.second;
-    if (!array_schema_->var_size(attr_name)) {
-      RETURN_NOT_OK(sub_partitioner.set_result_budget(
-          attr_name.c_str(), result_budget.size_fixed_));
-    } else {
-      RETURN_NOT_OK(sub_partitioner.set_result_budget(
-          attr_name.c_str(),
-          result_budget.size_fixed_,
-          result_budget.size_var_));
-    }
-  }
-
-  // Move to the first partition.
-  RETURN_NOT_OK(sub_partitioner.next(&read_state_.unsplittable_));
-
-  while (true) {
-    // If the sub-partitioners memory budget was too low, we may
-    // have been unable to split. In this scenario, double the
-    // budget and retry. In the worst-case scenario, the budget
-    // will equal the parent partitioner's budget.
-    while (read_state_.unsplittable_) {
-      uint64_t partitioner_memory_budget;
-      uint64_t partitioner_memory_budget_var;
-      RETURN_NOT_OK(partitioner->get_memory_budget(
-          &partitioner_memory_budget, &partitioner_memory_budget_var));
-
-      sub_partitioner_memory_budget = std::min(
-          partitioner_memory_budget, sub_partitioner_memory_budget * 2);
-      sub_partitioner_memory_budget_var = std::min(
-          partitioner_memory_budget_var, sub_partitioner_memory_budget_var * 2);
-
-      RETURN_NOT_OK(sub_partitioner.set_memory_budget(
-          sub_partitioner_memory_budget, sub_partitioner_memory_budget));
-
-      RETURN_NOT_OK(sub_partitioner.next(&read_state_.unsplittable_));
-    }
+  // Fetch the sub partitioner's memory budget.
+  bool found = false;
+  auto config = storage_manager_->config();
+  uint64_t cfg_sub_memory_budget = 0;
+  RETURN_NOT_OK(config.get<uint64_t>(
+      "sm.sub_partitioner_memory_budget", &cfg_sub_memory_budget, &found));
+  assert(found);
 
+  // The default value, 0, is used to bypass the sub-partitioner. The
```

Comment:
Although the condition is correct, I wouldn't specify in the comment what the default value is. The reason is that if we ever update the default, it'll be difficult to track this in the comments and update (it may lead to confusion).

Reply:
Good idea, done.